### PR TITLE
combine_user_tables

### PR DESCRIPTION
**Pull request by ulfgebhardt**
## 🍰 Pullrequest

Combines both user tables and periphery data into one table.

### Issues

- None

### Todo

- [x] Test with Production Data
- [x] Test Seed
- [x] Test with Seed Data

### CODE REVIEW

**Comment by ulfgebhardt** [comment]
@Mogge Can you try this and test the seed & with seeded data?
It works very well with the production data
❤️

File: database/migrations/0017-combine_user_tables.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+  // Drop column `group_id` since it contains uniform data which is not the same as the uniform data
+  // on login_users. Since we do not need this data anyway, we sjust throw it away.
+  await queryFn('ALTER TABLE `state_users` DROP COLUMN `group_id`;')
+
+  // Remove the unique constraint from the pubkey
+  // TODO - check for code impact
+  await queryFn('ALTER TABLE `state_users` DROP INDEX `public_key`;')
+
+  // Allow NULL on the `state_users` pubkey like it is allowed on `login_users`
+  // TODO remove the random key shenanigans
```
Comment (Mogge) [inline database/migrations/0017-combine_user_tables.ts:17]: I think this is done

File: database/migrations/0017-combine_user_tables.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  await queryFn('ALTER TABLE `state_users` MODIFY COLUMN `public_key` binary(32) DEFAULT NULL;')
+
+  // instead use a unique constraint for the email like on `login_users`
```
Comment (Mogge) [inline database/migrations/0017-combine_user_tables.ts:20]: Sure we want to remove the unique constraint?

File: database/src/seeds/users/bibi-bloxberg.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
 export const bibiBloxberg = {
+  loginUserId: 1,
```
Comment (Mogge) [inline database/src/seeds/users/bibi-bloxberg.ts:2]: What for?

**Review by Mogge — APPROVED** [review]
Great work. I tried seeding, login, transaction, create users and create GDD. Everything worked

File: database/migrations/0017-combine_user_tables.ts (hunk 2 above)
Comment (ulfgebhardt) [reply database/migrations/0017-combine_user_tables.ts:19]:
```
// instead use a unique constraint for the email like on `login_users`
  // therefore do not allow null on `email` anymore
  await queryFn(
    'ALTER TABLE `state_users` MODIFY COLUMN `email` varchar(255) COLLATE utf8mb4_unicode_ci NOT NULL;',
  )
  // TODO - check for code impact
  await queryFn('ALTER TABLE `state_users` ADD CONSTRAINT `email` UNIQUE KEY (`email`);')
  ```
  
  It actually adds an unique constraint, not removes it. It disallows NULL on EMail to enable an unique constraint

File: database/src/seeds/users/bibi-bloxberg.ts (hunk 3 above)
Comment (ulfgebhardt) [reply database/src/seeds/users/bibi-bloxberg.ts]: Good catch

**Pull request merged**